datumaro/components/dataset.py

```diff
@@ -713,10 +713,29 @@ def remove(self, id: str, subset: Optional[str] = None) -> None:
 
     def filter(self, expr: str, filter_annotations: bool = False,
             remove_empty: bool = False) -> Dataset:
+        """
+        Filters-out some dataset items or annotations, using a custom filter
+        expression.
+
+        Results are stored in-place. Modifications are applied lazily.
+
+        Args:
+            expr (str) - XPath-formatted filter expression
```

Comment:
IMO, we shouldn't duplicate the type information in the documentation. It doesn't communicate anything new to the user and creates extra work for developers. And in the case of complex types, the type annotation syntax is more precise than English.

Reply:
Agree, don't know, why I came to this.